### PR TITLE
Fixed a ConcurrentModificationException

### PR DESCRIPTION
**Pull request by zanzan23**
Fixed a ConcurrentModificationException caused by more threads that access to the methods removeAttribute and getAttributeNames

### CODE REVIEW

**Comment by saikiran939** [comment]
Is it not possible to use reentrant read/write locks rather than synchronization blocks ?

**Comment by zanzan23** [comment]
That was my first idea too, but because of the static methods in which to put synchronization I preferred to use the old synchronization blocks

File: modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereRequestImpl.java [hunk 1]
```diff
@@ -2049,8 +2055,9 @@ private static void load(HttpServletRequest request, Builder b) {
             s = e.nextElement();
             b.headers.put(s, request.getHeader(s));
         }
-
-        e = request.getAttributeNames();
+        synchronized (b.request) {
```
Comment (jfarcand) [inline modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereRequestImpl.java]: No sync is needed on a `get`, no?

Comment (zanzan23) [reply modules/cpr/src/main/java/org/atmosphere/cpr/AtmosphereRequestImpl.java]: If it is not possible that a thread invokes the load method while another thread modifies the attributes of the HttpServletRequest, the sync block in the load
get method is not needed. The sync blocks that are needed in my opinion is the ones in method getAttributeNames, setAttribute and removeAttribute.

**Pull request merged**